authz/middleware.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package authz
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/ant0ine/go-json-rest/rest"
+	"github.com/mendersoftware/go-lib-micro/requestlog"
+	"github.com/mendersoftware/go-lib-micro/rest_utils"
+)
+
+// AuthzMiddleware checks the authorization on a given request.
+// It retrieves the requested resource and action, and delegates the authz check to an Authorizer.
+type AuthzMiddleware struct {
```

Comment:
Sorry for commenting on this so late. IMO it would be better if authz/resfunc were public. In which case, it should be possible to remove `NewAuthzMiddleware` entirely.